src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RiskService.kt

```diff
@@ -151,20 +154,22 @@ internal class RiskService(
 
   private fun getLatestCompleteAssessment(assessmentsResponse: AssessmentsResponse) =
     assessmentsResponse.assessments
-      ?.filter { !it.dateCompleted.isNullOrEmpty() }
-      ?.sortedBy { LocalDateTime.parse(it.dateCompleted).toLocalDate() }
-      ?.reversed()
-      ?.firstOrNull { it.assessmentStatus == "COMPLETE" && it.superStatus == "COMPLETE" }
+      ?.map { it.copy(initiationDate = it.dateCompleted ?: DEFAULT_DATE_TIME_FOR_NULL_VALUE) }
+      ?.sortedBy { it.dateCompleted }
+      ?.lastOrNull() { it.assessmentStatus == "COMPLETE" && it.superStatus == "COMPLETE" }
 
   private fun isLatestAssessment(it: Assessment?) =
     (it?.laterCompleteAssessmentExists == false && it.laterWIPAssessmentExists == false && it.laterPartCompSignedAssessmentExists == false && it.laterSignLockAssessmentExists == false && it.laterPartCompUnsignedAssessmentExists == false)
 
   private fun datesMatch(
     latestAssessment: Assessment?,
     mainOffenceDate: LocalDate?,
-  ) = latestAssessment?.offenceDetails?.any {
-    it.offenceDate != null && (LocalDateTime.parse(it.offenceDate).toLocalDate() == mainOffenceDate)
-  } == true
+  ) = latestAssessment
+    ?.offenceDetails
+    ?.map { it.copy(offenceDate = it.offenceDate ?: DEFAULT_DATE_TIME_FOR_NULL_VALUE) }
+    ?.any {
+      LocalDateTime.parse(it.offenceDate).toLocalDate() == mainOffenceDate
```

Comment:
Don't need the map
We can do

.any { it.offenceDate == mainOffenceDate }

Reply:
I don't think this will work here. `offenceDate `is a LocalDateTime String, `mainOffenceDate `is LocalDate, so the conversion/mapping is needed. Correct me if I'm wrong :D

Reply:
OK.  OffenceDate ought to be a Date not a DateTime (not suggesting we change that now though).

I think in that case, it's OK to go back to the original code for this - it checked for offenceDate being null before parsing so wouldn't suffer from problems.

Reply:
Done

Reply:
It doesn't look like the reversion has come through.

Reply:
Should be fine now